v2/car.go

```diff
@@ -94,21 +106,38 @@ func (h Header) WriteTo(w io.Writer) (n int64, err error) {
 		return
 	}
 	n += wn
-	if err = writeUint64To(w, h.CarV1Offset); err != nil {
+	if err = binary.Write(w, binary.LittleEndian, h.CarV1Offset); err != nil {
 		return
 	}
 	n += uint64BytesSize
-	if err = writeUint64To(w, h.CarV1Size); err != nil {
+	if err = binary.Write(w, binary.LittleEndian, h.CarV1Size); err != nil {
 		return
 	}
 	n += uint64BytesSize
-	if err = writeUint64To(w, h.IndexOffset); err != nil {
+	if err = binary.Write(w, binary.LittleEndian, h.IndexOffset); err != nil {
 		return
 	}
 	n += uint64BytesSize
 	return
 }
 
-func writeUint64To(w io.Writer, v uint64) error {
-	return binary.Write(w, binary.LittleEndian, v)
+func (h Header) ReadFrom(r io.Reader) (n int64, err error) {
```

Comment:
once you start benchmarking all this, you will probably end up finding that reading the entire header as a single chunk of []byte would most likely be faster. It avoids extra Read calls, and allows working directly with slices of byte. Plus, the header is pretty small in terms of byte buffers, anyway.